src/main/kotlin/com/github/quiltservertools/ledger/commands/subcommands/ExportCommand.kt

```diff
@@ -0,0 +1,64 @@
+package com.github.quiltservertools.ledger.commands.subcommands
+
+import com.github.quiltservertools.ledger.Ledger
+import com.github.quiltservertools.ledger.actions.ActionType
+import com.github.quiltservertools.ledger.actionutils.ActionSearchParams
+import com.github.quiltservertools.ledger.commands.BuildableCommand
+import com.github.quiltservertools.ledger.commands.CommandConsts
+import com.github.quiltservertools.ledger.commands.arguments.SearchParamArgument
+import com.github.quiltservertools.ledger.database.DatabaseManager
+import com.github.quiltservertools.ledger.utility.LiteralNode
+import com.github.quiltservertools.ledger.utility.TextColorPallet
+import com.github.quiltservertools.ledger.utility.literal
+import com.github.quiltservertools.ledger.utility.translate
+import kotlinx.coroutines.launch
+import me.lucko.fabric.api.permissions.v0.Permissions
+import net.fabricmc.loader.api.FabricLoader
+import net.minecraft.server.command.CommandManager.literal
+import net.minecraft.server.command.ServerCommandSource
+import net.minecraft.text.TranslatableText
+import java.nio.file.Files
+import java.text.SimpleDateFormat
+import java.time.Instant
+import java.util.*
+import kotlin.io.path.pathString
+
+object ExportCommand : BuildableCommand {
+    override fun build(): LiteralNode {
+        return literal("export")
+            .requires(Permissions.require("ledger.commands.export", CommandConsts.PERMISSION_LEVEL))
+            .then(SearchParamArgument.argument(CommandConsts.PARAMS)
+                .executes {
+                    run(it.source, SearchParamArgument.get(it, CommandConsts.PARAMS))
+                })
+            .build()
+    }
+
+    @SuppressWarnings("BlockingMethodInNonBlockingContext")
+    private fun run(source: ServerCommandSource, params: ActionSearchParams): Int {
+        Ledger.launch {
+            source.sendFeedback("text.ledger.export.started".translate().setStyle(TextColorPallet.primary), false)
+            val results = DatabaseManager.searchActions(params, 0)
+            val actions = mutableListOf<ActionType>()
+            for (n in results.page..results.pages) {
+                results.actions.forEach {
+                    actions.add(it)
+                }
+            }
+            source.sendFeedback(TranslatableText("text.ledger.export.actions", results.pages.toString().literal()
+                .setStyle(TextColorPallet.secondaryVariant)).setStyle(TextColorPallet.secondary), false)
+
+            val time = SimpleDateFormat("ss-mm-HH-dd-MM-yyyy").format(Date.from(Instant.now()))
+            val path = FabricLoader.getInstance().gameDir.resolve("ledger-export-${time}.txt")
+            var string = ""
+            actions.forEach {
+                string += "${it.getMessage().string}\n"
+            }
+            Files.createFile(path)
+            Files.writeString(path, string)
```

Comment:
Is it worth catching `IOException`s here to inform the user if something goes wrong saving the export to disk?